src/iuse.cpp

```diff
@@ -5646,7 +5651,10 @@ void iuse::play_music( player * const p, const tripoint &source, int const volum
 int iuse::mp3_on(player *p, item *it, bool t, const tripoint &pos)
 {
     if (t) { // Normal use
-        play_music( p, pos, 0, 50 );
+        if (p->has_item(it)) {
+            // player has earbuds on, we can listen
+            play_music( p, pos, 5, 50 );
```

Comment:
Any reason you changed the volume to 5? Ah, I see: so you can hear it during rain.

@kevingranade This conflicts with the ambient noise from the weather. The character should always be able to listen to their MP3 player (assuming they wear the earbuds), but even drizzling reduces the effective range of the sound. if the MP3 player uses volume=0, it will not be heard by the character during slightly bad weather. If it uses a larger volume, it will be heard by nearby Creatures as well.

Another bug: `player::can_hear` does not respect the weather based hearing reduction. `player::can_hear` returns true (and the music morale is applied), but the sounds class will not print the music description because the bad weather prevents the character from actually hearing it (so thinks the sounds class).

I suggest a special case for sounds having volume 0 and coming from the same tile as the player: noise from weather should be ignored there.